### PR TITLE
Rename Label `cluster` to `storage_cluster_id` for apiserver_storage_size_bytes metric

### PR DESCRIPTION
**Pull request by dims**
Fix for https://github.com/kubernetes/kubernetes/issues/124277

`cluster` is an overloaded term in the observability space. This is an attempt to ensure we are very specific about the meaning of the label we emit by picking a new name.

#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
For apiserver_storage_size_bytes metric, we are renaming the label for etcd to be "storage_cluster_id" instead of "cluster" to to reduce conflict and be very specific.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig instrumentation

### CODE REVIEW

**Comment by logicalhan** [comment]
I think we have to backport this if we fix this, since this metric is stable, the API should be consistent across versions..

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Comment by dims** [comment]
@logicalhan @serathius do we want this in 1.30 itself? please add `/priority-critical-urgent` and `/milestone v1.30` if so.

(had to fix one more linter thing, so needs `/lgtm` as well)

**Comment by dims** [comment]
/assign @logicalhan @dashpole @serathius @dgrisonnet @rexagod

> This PR [may require stable metrics review](https://git.k8s.io/community/contributors/devel/sig-instrumentation/metric-stability.md).

Looks like we need 2 sig-instrumentation approvers here.

**Pull request merged**
